examples/simple_trainer_mcmc.py

```diff
@@ -644,17 +644,17 @@ def add_new_gs(self, cap_max):
 
     @torch.no_grad()
     def add_noise_to_gs(self, last_lr):
-        L = build_scaling_rotation(
-            torch.exp(self.splats["scales"]), self.splats["quats"]
-        )
-        actual_covariance = L @ L.transpose(1, 2)
+        opacities = torch.sigmoid(self.splats["opacities"])
+        scales = torch.exp(self.splats["scales"])
+        quats_normalized = F.normalize(self.splats["quats"], dim=-1)
+        actual_covariance = scale_rot_to_cov3d(scales, 1.0, quats_normalized)
```

Comment:
I think you can just use 
```
from gsplat import quat_scale_to_covar_preci
actual_covariance, _ = quat_scale_to_covar_preci(self.splats["quats"], scales, compute_covar=True, compute_perci=False, triu=False)
```

`quat_scale_to_covar_preci` is with fused CUDA implementation so should be faster. Also it does normalization on quats internally so there is no need to explicitly call `F.normalize`. We plan to deprecate the support of `cuda_legacy` at some point in the future so I would try to avoid relying on functions from that.